src/wp-admin/site-editor.php

```diff
@@ -30,7 +30,7 @@
 $home_template = _resolve_home_block_template();
 if ( $home_template && empty( $_GET['postType'] ) && empty( $_GET['postId'] ) ) {
 	$redirect_url = add_query_arg(
-		$home_template,
+		array_merge( $home_template, $_GET ),
```

Comment:
I went with a general fix. But I can update this only to support the `styles` query param.

Reply:
Thanks, @peterwilsoncc.

We can land general fix in the next major release.